src/cladetime/sequence.py

```diff
@@ -298,3 +300,61 @@ def parse_sequence_assignments(df_assignments: pl.DataFrame) -> pl.DataFrame:
     df_assignments = df_assignments.insert_column(1, seq)  # type: ignore
 
     return df_assignments
+
+
+def filter_sequence_data(sequence_ids: set, url_sequence: str, output_path: Path) -> tuple[Path, int, int]:
+    """Filter a fasta file against a specific set of sequences.
+
+    Download a sequence file (in FASTA format) from Nexstrain, filter
+    it against a set of specific sequence ids (GenBank accession numbers),
+    and write the filtered sequences to a new file.
+
+    Parameters
+    ----------
+    sequence_ids : set
+        GenBank accession numbers used to filter the sequence file
+    url_sequence : str
+        The URL to a file of SARS-CoV-2 GenBank sequences published by Nexstrain.
+        The file is should be in .fasta format using the lzma compression
+        method (e.g., "https://data.nextstrain.org/files/ncov/open/100k/sequences.fasta.xz")
+    output_path : pathlib.Path
+        Where to save the filtered sequence file
+
+    Returns
+    -------
+    Tuple[pathlib.Path, int, int]
+        A tuple containing the full path to the filtered sequence file, the
+        number of original sequences, and the number of filtered sequences
+    """
+    session = _get_session()
+
+    # FIXME: validate url_sequence (should be in filename.fasta.xz format)
+    # alternately, we could expand this function to handle other types
+    # of compression schemas (ZSTD) or none at all
+
+    # download the original sequence file
+    logger.info("Staring sequence file download", url=url_sequence)
```

Comment:
Staring --> starting (github won't let me suggest changes?) [i almost didn't bother typing this comment]

Reply:
Re: the missing suggest changes button. I noticed the same thing when reviewing your sarix PR. It seems like that's a drawback to reviewing commit by commit (a pattern which I otherwise really like): https://github.com/orgs/community/discussions/7128